scylla-cdc/src/log_reader.rs

```diff
@@ -68,7 +68,7 @@ impl CDCReaderWorker {
         let mut stream_reader_tasks = FuturesUnordered::new();
 
         let mut next_generation: Option<GenerationTimestamp> = None;
-        let mut had_first_generation: bool = false;
+        let mut last_fetched_generation: Option<GenerationTimestamp> = None;
```

Comment:
I think `current_generation` would be a more appropriate name. It's not the generation you have fetched most recently (next_generation fits this description better), but rather the one we are reading at the moment.